geospaas_processing/copiers.py

```diff
@@ -0,0 +1,72 @@
+import os
+import shutil
+import logging
+
+import django
+
+from geospaas.catalog.models import Dataset
+from geospaas.catalog.managers import LOCAL_FILE_SERVICE
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Copier():
```

Comment:
```suggestion
class CopyManager():
```
Maybe? To be consistent with DownloadManagers

Reply:
copying action is one type of action, unlike the download action which uses an specific kind of downloader and manages by download manager. It is better to keep this name in hormony with downloader.py. Respectfully, what is your opinion @aperrin66 ? Am I understand the downloader.py correctly? Am I concluding the correct way from my understanding?

Reply:
I'm not sure either of your solutions is well suited for this case.
We use Dowloaders and a DownloadManager because we need different implementations of the downloading action depending on the type of dataset. This is not the case here, because copying is always done the same way (or maybe you could have one copier for "real" copies, and one for symlinks, but that seems a little over the top).

To be honest, I'm not even sure if object oriented programming is necessary here.

Reply:
I just had a thought: couldn't the copying mechanism be implemented as a Downloader? It's just a particular case were the files are located on the same server. This would remove the need for a separate CLI, config file, etc...

Reply:
Defining a new workflow for differentiating the symlink from file itself is not necessary because there is only one line of code which varies between them! only one line! So, there is no need to do that.

Yeah, you are right about the OOP. I have done this task with some functions that are only gathered together in a class without any constructer! Then I saw that the arguments of the functions are a lot and they are repeat several times, so I decided to you use OOP for `simplification of implementation`.

By the way, OOP helps us to make sure that all of these:
```python
        self.type_in_flag_file = type_in_flag_file
        self.destination_path = destination_path
        self.flag_file_request = flag_file_request
        self.link_request = link_request
        self.datasets = Dataset.objects.filter(**criteria)
```
Are not changing during one copy action! This makes it robust and less error prone for the workflow! It is because they are constant during the execution of functions!



Regarding https://github.com/nansencenter/django-geo-spaas-processing/pull/24#discussion_r529387018:
This is one of those thunderbolt ideas that needs time and effort! I think it worth it. BUT please let us finish this PR and make a version number after it. And then we will implement this idea! OK?

I also appreciated if you two tell to how to name that class!!!  :)

Reply:
Maybe it is better for me to write that attributes in the constructor with capitalised letters based on the pep8.